### PR TITLE
Remove circular dependency between fetcher(uptane) and packagemanager

### PR DESCRIPTION
**Pull request by doanac**
This cleans up the relationship between the fetcher, sotauptaneclient, and packagemanager. It was previously an RFC #1135 but has been cleaned up and rebased on master.

### CODE REVIEW

**Comment by doanac** [comment]
my clang tidy isn't quite the same, but i'll try and address these failures quickly

File: src/libaktualizr/package_manager/androidmanager.h [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 
 class AndroidInstallationDispatcher;
 
 class AndroidManager : public PackageManagerInterface {
  public:
-  explicit AndroidManager(std::shared_ptr<INvStorage> storage, std::shared_ptr<Bootloader> bootloader)
-      : storage_(std::move(storage)), bootloader_(std::move(bootloader)) {}
+  explicit AndroidManager(PackageConfig pconfig, std::shared_ptr<INvStorage> storage,
+                          std::shared_ptr<Bootloader> bootloader, std::shared_ptr<HttpInterface> http)
+      : PackageManagerInterface(std::move(pconfig), std::move(storage), std::move(bootloader), std::move(http)) {}
```
[the remark this answers is not in the document]
Comment (doanac) [reply src/libaktualizr/package_manager/androidmanager.h]: good catch. fixed now.

File: src/libaktualizr/package_manager/ostreemanager.cc [hunk 2]
```diff
@@ -223,14 +223,22 @@ data::InstallationResult OstreeManager::finalizeInstall(const Uptane::Target &ta
...
   if (sysroot_smart == nullptr) {
     throw std::runtime_error("Could not find OSTree sysroot at: " + config.sysroot.string());
   }
 }
 
+bool OstreeManager::fetchTarget(const Uptane::Target &target, Uptane::Fetcher &fetcher, const KeyManager &keys,
+                                FetcherProgressCb progress_cb, const api::FlowControlToken *token) {
+  if (!target.IsOstree()) {
+    return PackageManagerInterface::fetchTarget(target, fetcher, keys, progress_cb, token);
+  }
```
[the remark this answers is not in the document]
Comment (eu-smirnov) [reply src/libaktualizr/package_manager/ostreemanager.cc]: @patrickvacek I still don't quite understand how do you want to support a case with an OSTree primary and kVirtual secondaries? SotaUptaneClient has only one package manager, `OstreeManager` in this case, which method gets called in `SotaUptaneClient::downloadImage`.
As an alternative to this `if (!target.IsOstree())` I see to assign each secondary a package manager and call their methods in `downloadImage`.
Or am I completely missing something?

**Comment by eu-smirnov** [comment]
I have some questions on how it all should look like in the end, but this PR seems to be a step in the right direction to me and it doesn't interfere with any changes that I was doing to the fetcher code recently. As it touches the code in lots of different places, I would probably merge it sooner and do the changes later if necessary.

File: src/libaktualizr/package_manager/ostreemanager.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (patrickvacek) [reply src/libaktualizr/package_manager/ostreemanager.cc]: @eu-smirnov No, you are correct. My bad, I was being overzealous. You are correct, this is the one download manager for primary and secondaries, so of course it has to be able to handle anything. Still worth checking if the backend lets us try to install binaries on an OSTree system, though.

**Pull request merged**